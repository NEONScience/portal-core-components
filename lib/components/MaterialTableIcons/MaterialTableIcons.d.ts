export default MaterialTableIcons;
declare namespace MaterialTableIcons {
    const Add: React.ForwardRefExoticComponent<React.RefAttributes<any>>;
    const Check: React.ForwardRefExoticComponent<React.RefAttributes<any>>;
    const Clear: React.ForwardRefExoticComponent<React.RefAttributes<any>>;
    const Delete: React.ForwardRefExoticComponent<React.RefAttributes<any>>;
    const DetailPanel: React.ForwardRefExoticComponent<React.RefAttributes<any>>;
    const Edit: React.ForwardRefExoticComponent<React.RefAttributes<any>>;
    const Export: React.ForwardRefExoticComponent<React.RefAttributes<any>>;
    const Filter: React.ForwardRefExoticComponent<React.RefAttributes<any>>;
    const FirstPage: React.ForwardRefExoticComponent<React.RefAttributes<any>>;
    const LastPage: React.ForwardRefExoticComponent<React.RefAttributes<any>>;
    const NextPage: React.ForwardRefExoticComponent<React.RefAttributes<any>>;
    const PreviousPage: React.ForwardRefExoticComponent<React.RefAttributes<any>>;
    const ResetSearch: React.ForwardRefExoticComponent<React.RefAttributes<any>>;
    const Search: React.ForwardRefExoticComponent<React.RefAttributes<any>>;
    const SortArrow: React.ForwardRefExoticComponent<React.RefAttributes<any>>;
    const ThirdStateCheck: React.ForwardRefExoticComponent<React.RefAttributes<any>>;
    const ViewColumn: React.ForwardRefExoticComponent<React.RefAttributes<any>>;
}
import React from "react";
