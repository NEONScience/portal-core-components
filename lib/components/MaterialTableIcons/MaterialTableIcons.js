"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _AddBox = _interopRequireDefault(require("@material-ui/icons/AddBox"));
var _ArrowUpward = _interopRequireDefault(require("@material-ui/icons/ArrowUpward"));
var _Check = _interopRequireDefault(require("@material-ui/icons/Check"));
var _ChevronLeft = _interopRequireDefault(require("@material-ui/icons/ChevronLeft"));
var _ChevronRight = _interopRequireDefault(require("@material-ui/icons/ChevronRight"));
var _Clear = _interopRequireDefault(require("@material-ui/icons/Clear"));
var _DeleteOutline = _interopRequireDefault(require("@material-ui/icons/DeleteOutline"));
var _Edit = _interopRequireDefault(require("@material-ui/icons/Edit"));
var _FilterList = _interopRequireDefault(require("@material-ui/icons/FilterList"));
var _FirstPage = _interopRequireDefault(require("@material-ui/icons/FirstPage"));
var _LastPage = _interopRequireDefault(require("@material-ui/icons/LastPage"));
var _Remove = _interopRequireDefault(require("@material-ui/icons/Remove"));
var _SaveAlt = _interopRequireDefault(require("@material-ui/icons/SaveAlt"));
var _Search = _interopRequireDefault(require("@material-ui/icons/Search"));
var _TableChart = _interopRequireDefault(require("@material-ui/icons/TableChart"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); } // This exists so that Material Table doesn't have to rely on loading the Material Icon font in
// the HTML <HEAD>. See https://github.com/mbrn/material-table/issues/51#issuecomment-508384214
const MaterialTableIcons = {
  Add: /*#__PURE__*/(0, _react.forwardRef)((props, ref) => /*#__PURE__*/_react.default.createElement(_AddBox.default, _extends({}, props, {
    ref: ref
  }))),
  Check: /*#__PURE__*/(0, _react.forwardRef)((props, ref) => /*#__PURE__*/_react.default.createElement(_Check.default, _extends({}, props, {
    ref: ref
  }))),
  Clear: /*#__PURE__*/(0, _react.forwardRef)((props, ref) => /*#__PURE__*/_react.default.createElement(_Clear.default, _extends({}, props, {
    ref: ref
  }))),
  Delete: /*#__PURE__*/(0, _react.forwardRef)((props, ref) => /*#__PURE__*/_react.default.createElement(_DeleteOutline.default, _extends({}, props, {
    ref: ref
  }))),
  DetailPanel: /*#__PURE__*/(0, _react.forwardRef)((props, ref) => /*#__PURE__*/_react.default.createElement(_ChevronRight.default, _extends({}, props, {
    ref: ref
  }))),
  Edit: /*#__PURE__*/(0, _react.forwardRef)((props, ref) => /*#__PURE__*/_react.default.createElement(_Edit.default, _extends({}, props, {
    ref: ref
  }))),
  Export: /*#__PURE__*/(0, _react.forwardRef)((props, ref) => /*#__PURE__*/_react.default.createElement(_SaveAlt.default, _extends({}, props, {
    ref: ref
  }))),
  Filter: /*#__PURE__*/(0, _react.forwardRef)((props, ref) => /*#__PURE__*/_react.default.createElement(_FilterList.default, _extends({}, props, {
    ref: ref
  }))),
  FirstPage: /*#__PURE__*/(0, _react.forwardRef)((props, ref) => /*#__PURE__*/_react.default.createElement(_FirstPage.default, _extends({}, props, {
    ref: ref
  }))),
  LastPage: /*#__PURE__*/(0, _react.forwardRef)((props, ref) => /*#__PURE__*/_react.default.createElement(_LastPage.default, _extends({}, props, {
    ref: ref
  }))),
  NextPage: /*#__PURE__*/(0, _react.forwardRef)((props, ref) => /*#__PURE__*/_react.default.createElement(_ChevronRight.default, _extends({}, props, {
    ref: ref
  }))),
  PreviousPage: /*#__PURE__*/(0, _react.forwardRef)((props, ref) => /*#__PURE__*/_react.default.createElement(_ChevronLeft.default, _extends({}, props, {
    ref: ref
  }))),
  ResetSearch: /*#__PURE__*/(0, _react.forwardRef)((props, ref) => /*#__PURE__*/_react.default.createElement(_Clear.default, _extends({}, props, {
    ref: ref
  }))),
  Search: /*#__PURE__*/(0, _react.forwardRef)((props, ref) => /*#__PURE__*/_react.default.createElement(_Search.default, _extends({}, props, {
    ref: ref
  }))),
  SortArrow: /*#__PURE__*/(0, _react.forwardRef)((props, ref) => /*#__PURE__*/_react.default.createElement(_ArrowUpward.default, _extends({}, props, {
    ref: ref
  }))),
  ThirdStateCheck: /*#__PURE__*/(0, _react.forwardRef)((props, ref) => /*#__PURE__*/_react.default.createElement(_Remove.default, _extends({}, props, {
    ref: ref
  }))),
  ViewColumn: /*#__PURE__*/(0, _react.forwardRef)((props, ref) => /*#__PURE__*/_react.default.createElement(_TableChart.default, _extends({}, props, {
    ref: ref
  })))
};
var _default = exports.default = MaterialTableIcons;