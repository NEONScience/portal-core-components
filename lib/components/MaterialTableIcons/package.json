{
  "private": true,
  "name": "material-table-icons",
  "main": "./MaterialTableIcons.jsx",
  "module": "./MaterialTableIcons.js"
}
