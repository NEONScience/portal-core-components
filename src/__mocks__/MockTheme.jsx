import React from 'react';

import CssBaseline from '@mui/material/CssBaseline';
import { StyledEngineProvider, ThemeProvider } from '@mui/material/styles';

import Theme from '../lib_components/components/Theme/Theme';

const MockTheme = ({ children }) => {
  return (
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={Theme}>
        <CssBaseline />
        {children}
      </ThemeProvider>
    </StyledEngineProvider>
  );
}

export default MockTheme;
