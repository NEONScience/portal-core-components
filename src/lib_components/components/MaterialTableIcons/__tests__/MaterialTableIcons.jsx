import React from 'react';
import renderer from 'react-test-renderer';

import MockTheme from '../../../../__mocks__/MockTheme';
import MaterialTableIcons from '../MaterialTableIcons';

describe('MaterialTableIcons', () => {
  Object.keys(MaterialTableIcons).forEach((icon) => {
    test(`renders ${icon} icon correctly`, () => {
      const Icon = MaterialTableIcons[icon];
      const tree = renderer.create(<MockTheme><Icon /></MockTheme>).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
