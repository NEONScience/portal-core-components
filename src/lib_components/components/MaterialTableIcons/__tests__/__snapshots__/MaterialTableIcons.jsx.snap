// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaterialTableIcons renders Add icon correctly 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="AddBoxIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2m-2 10h-4v4h-2v-4H7v-2h4V7h2v4h4z"
  />
</svg>
`;

exports[`MaterialTableIcons renders Check icon correctly 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="CheckIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
  />
</svg>
`;

exports[`MaterialTableIcons renders Clear icon correctly 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="ClearIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
  />
</svg>
`;

exports[`MaterialTableIcons renders Delete icon correctly 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="DeleteOutlineIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM8 9h8v10H8zm7.5-5-1-1h-5l-1 1H5v2h14V4z"
  />
</svg>
`;

exports[`MaterialTableIcons renders DetailPanel icon correctly 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="ChevronRightIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
  />
</svg>
`;

exports[`MaterialTableIcons renders Edit icon correctly 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="EditIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75z"
  />
</svg>
`;

exports[`MaterialTableIcons renders Export icon correctly 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="SaveAltIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M19 12v7H5v-7H3v7c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2v-7zm-6 .67 2.59-2.58L17 11.5l-5 5-5-5 1.41-1.41L11 12.67V3h2z"
  />
</svg>
`;

exports[`MaterialTableIcons renders Filter icon correctly 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="FilterListIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M10 18h4v-2h-4zM3 6v2h18V6zm3 7h12v-2H6z"
  />
</svg>
`;

exports[`MaterialTableIcons renders FirstPage icon correctly 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="FirstPageIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M18.41 16.59 13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
  />
</svg>
`;

exports[`MaterialTableIcons renders LastPage icon correctly 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="LastPageIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M5.59 7.41 10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
  />
</svg>
`;

exports[`MaterialTableIcons renders NextPage icon correctly 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="ChevronRightIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
  />
</svg>
`;

exports[`MaterialTableIcons renders PreviousPage icon correctly 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="ChevronLeftIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
  />
</svg>
`;

exports[`MaterialTableIcons renders ResetSearch icon correctly 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="ClearIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
  />
</svg>
`;

exports[`MaterialTableIcons renders Search icon correctly 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="SearchIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
  />
</svg>
`;

exports[`MaterialTableIcons renders SortArrow icon correctly 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="ArrowUpwardIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="m4 12 1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8z"
  />
</svg>
`;

exports[`MaterialTableIcons renders ThirdStateCheck icon correctly 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="RemoveIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M19 13H5v-2h14z"
  />
</svg>
`;

exports[`MaterialTableIcons renders ViewColumn icon correctly 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="TableChartIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M10 10.02h5V21h-5zM17 21h3c1.1 0 2-.9 2-2v-9h-5zm3-18H5c-1.1 0-2 .9-2 2v3h19V5c0-1.1-.9-2-2-2M3 19c0 1.1.9 2 2 2h3V10H3z"
  />
</svg>
`;
